core/commands/name/publish.go

```diff
@@ -128,23 +128,17 @@ Alternatively, publish an <ipfs-path> using a valid PeerID (as listed by
 			}
 		}
 
-		out, err := api.Name().Publish(req.Context, p, opts...)
+		name, err := api.Name().Publish(req.Context, p, opts...)
 		if err != nil {
 			if err == iface.ErrOffline {
 				err = errAllowOffline
 			}
 			return err
 		}
 
-		// parse path, extract cid, re-base cid, reconstruct path
-		pid, err := peer.Decode(out.Name())
-		if err != nil {
-			return err
-		}
-
 		return cmds.EmitOnce(res, &IpnsEntry{
-			Name:  keyEnc.FormatID(pid),
-			Value: out.Value().String(),
+			Name:  strings.TrimPrefix(name.String(), "/ipns/"),
```

Comment:
I did not want to keep this. But I'm keeping for two reasons:
1. `ipns name publish` command keeps outputting the same expected output with just the Peer ID (note that it is now normalized).
2. interop with js-ipfs pubsub ipns tests fail if we change this because it uses the old RPC API

We can fix 1 and 2 though if that's desirable.

Reply:
Hm..  am kinda confused why we ended up with prefix as part of Name string. 
Is it because it was mentioned in https://specs.ipfs.tech/ipns/ipns-record/#ipns-name? 
Lets remove it – see https://github.com/ipfs/boxo/pull/339#discussion_r1229847123